### PR TITLE
193: JsonPatchHelper should normalize URIs when using EMF-style paths

### LINKED ISSUE #193: API V2: JsonPatchHelper should normalize URIs when using EMF-style paths
When using EMF-style paths with Json Patch, we include the resource URI in the path. Unlike the ?modeluri query, this URI is not normalized, which may cause some issues.

We should add URI normalization to JsonPatchHelper

### PR DESCRIPTION
**Pull request by CamilleLetavernier**
- Normalize URIs in ~DefaultModelResourceManager~ and JsonPatchHelper

Contributed on behalf of STMicroelectronics.

fixes #193

### CODE REVIEW

File: tests/org.eclipse.emfcloud.modelserver.emf.tests/src/org/eclipse/emfcloud/modelserver/emf/common/watchers/FileModelWatcherIntegrationTest.java [hunk 1]
```diff
@@ -151,8 +152,8 @@ protected void configure() {
             bind(ModelRepository.class).to(DefaultModelRepository.class).in(Singleton.class);
             bind(ModelResourceManager.class).to(DefaultModelResourceManager.class).in(Singleton.class);
             bind(ResourceSetFactory.class).to(DefaultResourceSetFactory.class).in(Singleton.class);
-            bind(URIConverter.class).to(DefaultModelURIConverter.class).in(Singleton.class);
-
+            bind(ModelURIConverter.class).to(DefaultModelURIConverter.class).in(Singleton.class);
```
Comment (cdamus) [inline tests/org.eclipse.emfcloud.modelserver.emf.tests/src/org/eclipse/emfcloud/modelserver/emf/common/watchers/FileModelWatcherIntegrationTest.java:155]: Thanks for that!

**Review by cdamus — APPROVED** [review]
LGTM!

**Comment by CamilleLetavernier** [comment]
I've reverted the changes to DefaultModelResourceManager, because it seems that all URIs should be normalized before they get passed to this class. Adding the ModelURIConverter to the DefaultModleResourceManager seemed to add unnecessary complexity (Especially for test setup), and was mostly redundant.

**Comment by cdamus** [comment]
> I've reverted the changes to DefaultModelResourceManager, because it seems that all URIs should be normalized before they get passed to this class. Adding the ModelURIConverter to the DefaultModleResourceManager seemed to add unnecessary complexity (Especially for test setup), and was mostly redundant.

I think I reviewed this after you made that change? Because I didn't notice it and I would have commented that yes, it was unnecessary. 😉

**Review by sgraband — APPROVED** [review]
Thanks @CamilleLetavernier! LGTM :+1:

**Pull request merged**